apps/cli/lib/sync.ex

```diff
@@ -0,0 +1,101 @@
+defmodule CLI.Sync do
+  @moduledoc """
+  Tools to sync new blocks.
+  """
+  require Logger
+
+  alias Blockchain.{Blocktree, Chain}
+  alias CLI.State
+  alias MerklePatriciaTree.DB
+
+  @type block_limit :: integer() | :infinite
+
+  @save_block_interval 100
+
+  @doc """
+  Recursively adds blocks to a tree. This function will
+  run forever unless `max_new_blocks` is set, in which
+  case it will add that many blocks and then return.
+  """
+  @spec sync_new_blocks(
+          module(),
+          any(),
+          DB.db(),
+          Chain.t(),
+          Blocktree.t(),
+          block_limit()
+        ) :: {:ok, Blocktree.t()} | {:error, any()}
+  def sync_new_blocks(
+        block_provider,
+        block_provider_state,
+        db,
+        chain,
+        tree,
+        block_number,
+        block_limit \\ :infinite,
+        highest_known_block_number \\ nil
+      ) do
+    case should_continue?(block_limit) do
+      :stop ->
+        {:ok, tree}
+
+      {:continue, next_block_limit} ->
```

Comment:
might be good to extract or refactor this?

Reply:
Right now, I just wanted some progress to let this run indefinitely or exit. I think it's worth refactoring this once we realize how P2P syncing will work.